streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1081,8 +1081,9 @@ private void shutdown(final boolean cleanRun) {
         streamsMetrics.removeAllSensors();
     }
 
+    // visible for testing
     @SuppressWarnings("ThrowableNotThrown")
-    private void shutdownTasksAndState(final boolean cleanRun) {
+    void shutdownTasksAndState(final boolean cleanRun) {
```

Comment:
We can. But I don't think reflections are any better -- in the opposite, it adds noise to the test code. Also, we apply this pattern all over the code base and never use reflections, thus, we we change it, we should change everywhere \cc @dguy